### DISCUSSION TITLE
Question about gen_vx_mask

### DISCUSSION THREAD

**Discussion by junmeiban** [dtcenter/METplus#1463] [Configuration] [answered]
Dear MET-help,

I am using the grid_stat function to verify precipitation forecasts from MPAS against CMORPH precipitation product. 

I would like to verify precipitation between -60 ~60 latitude over land or ocean separately. 

I checked the “met-10.1.0-beta3/data/poly/” directory, but It looks like there is no land_sea mask file. So I am using gen_vx_mask function, and get the following errors:

``` 
>gen_vx_mask -type data latlon-landmask1.nc latlon-landmask1.nc out.nc -mask_field landmask 

DEBUG 1: Input File:		[latlon-landmask1.nc]

DEBUG 1: Mask File:		[latlon-landmask1.nc]

DEBUG 2: Parsed Data Grid:	LatLon (720 x 360)

ERROR  :

ERROR  : yyerror() -> syntax error in file "/tmp/met_config_19238_0"

ERROR  :

ERROR  :    line   = 2

ERROR  :

ERROR  :    column = 8

ERROR  :

ERROR  :    text   = "(nul)"

ERROR  :

ERROR  :

ERROR  : (nul)

ERROR  : (nul)

ERROR  :
```
(1)Do you have sample input_file and mask_file in netcdf format for testing gen_vx_mask function on Cheyenne? If so, I can check the correctness of my input/mask files.

(2)Would you please correct my gen_vx_mask command?

Many thanks for your help,

Junmei

**Comment by junmeiban** [comment, marked as answer]
Thanks @byoung-joo so much for helping me about this issue:
(1)My input nc file is not in CF-compliant format, so it can not read correctly.
(2)Use gfs grib file as input file and mask file.
(3)Use -type data to mask land or sea, then, use -type lat to mask -60~60 latitude.
I can got the correct mask file. This issue can be closed. Thanks!

**Reply by JohnHalleyGotway** [reply]
@junmeiban, I see you have a question about using the Gen-Vx-Mask tool to generate two "complex" masks:
1. All land grid points between -60 and 60 latitude.
2. All water grid points between -60 and 60 latitude.

While I don't have your sample NetCDF file available for testing, I did use a sample GRIB2 GFS file to demonstrate how you could do this. So I'll use a file named "gfs.grib2" in my examples below.

I'd recommend that your run Gen-Vx-Mask 3 times:

1. Define the -60 to 60 latitude mask using "-type lat" option:
```
gen_vx_mask gfs.grib2 gfs.grib2 lat_mask.nc -type lat -thresh '>=-60&&<=60'
```

2. Intersect the latitude mask with a land mask using the "-type data" option:
```
gen_vx_mask lat_mask.nc gfs.grib2 land_mask.nc -type data -mask_field 'name="LAND"; level="L0";' -thresh eq1 -intersection
```

3. Intersect the latitude mask with a water mask using the "-type data" option:
```
gen_vx_mask lat_mask.nc gfs.grib2 water_mask.nc -type data -mask_field 'name="LAND"; level="L0";' -thresh eq0 -intersection
```

Using ncview to visualize the output, here's what these 2 masks look like:
<img width="723" alt="Screen Shot 2022-03-02 at 3 14 37 PM" src="https://user-images.githubusercontent.com/21087144/156458720-0d841fa1-85d5-4b67-a27d-c37ce1f609c9.png">

I'd recommend using the same approach with your NetCDF file.
In my examples, I read the land/sea mask from the GRIB2 file using 'name="LAND"; level="L0";'
When you switch in your NetCDF file, you'll use something like this 'name="VarName"; level="(\*,\*)";'
Where VarName is the variable name containing your land mask data.

Hopefully this example helps get you going!

**Comment by junmeiban** [comment]
Hi John,
I can generate mask files following your steps. So you already answered  my question. 
Many thanks for your help.
Junmei

**Reply by JohnHalleyGotway** [reply]
Great! Thanks for confirming. I'll go ahead and mark this discussion as being answered and lock the conversation on it. If other questions come up, please feel free to create a new discussion.
